wgengine/magicsock/magicsock_linux.go

```diff
@@ -288,3 +291,24 @@ func setBPF(conn net.PacketConn, filter []bpf.RawInstruction) error {
 	}
 	return nil
 }
+
+// trySetSocketBuffer sets SO_SNDBUF and SO_RECVBUF on pconn to socketBufferSize,
+// logging an error if it occurs.
+func trySetSocketBuffer(pconn nettype.PacketConn, logf logger.Logf) {
+	if c, ok := pconn.(*net.UDPConn); ok {
+		rc, err := c.SyscallConn()
+		if err != nil {
+			logf("magicsock: failed to get raw syscall conn: %v, falling back to portable interface", err)
+			portableTrySetSocketBuffer(pconn, logf)
+			return
+		}
+		rc.Control(func(fd uintptr) {
+			if err := syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_RCVBUFFORCE, socketBufferSize); err != nil {
+				logf("magicsock: failed to force-set UDP read buffer size to %d: %v", socketBufferSize, err)
```

Comment:
Fall back to portable impl if these sets fail?

I guess it depends how old the FORCE sockopts are. Are they liable to not work on some of the older kernels we support (e.g. olde synology) ?

Reply:
they require CAP_NET_ADMIN